src/torch_models.py

```diff
@@ -482,6 +505,43 @@ def _predict(self, x: Array) -> Array:
         sample_idx = torch.argmax(scores)
         return sample_ys[sample_idx]
 
+    def _predict_derivative_free(self, x: Array) -> Array:
+        # Reference: https://arxiv.org/pdf/2109.00137.pdf (Algorithm 1).
+        # This method reportedly works well in up to 5 dimensions.
+        # Since we are using torch for random sampling, and since we want
+        # to ensure deterministic predictions, we need to reseed torch.
```

Comment:
clarify that we can't seed in the init because we want consistency across multiple calls to this predictor for the same input

Reply:
(i think this is what's happening in the option model stuff)